### PR TITLE
Added Microservice builder

### PR DESCRIPTION
**Pull request by sfisol**

### CODE REVIEW

File: src/server/builder.rs [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+        D: AppDataWrapper + 'static,
+        T: StatsPresenter<D> + 'static + Clone + Send + Sync,
+        F: Fn(&mut web::ServiceConfig) + Send + Clone + 'static + Sized,
+        C: Fn() -> Cors + Send + Clone + 'static,
+    {
+        if let Some(cors_factory) = cors_factory {
+            self.start_inner(prepare_app_data, configure_app, cors_factory).await
+        } else {
+            self.start_inner(prepare_app_data, configure_app, default_cors).await
+        }
```
Comment (damszew) [inline src/server/builder.rs:103]: consider extracting cors config to variable, such that:
```suggestion
        let cors = cors_factory.unwrap_or(default_cors);
        self.start_inner(prepare_app_data, configure_app, cors).await
```

File: src/server/builder.rs [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+        C: Fn() -> Cors + Send + Clone + 'static,
+    {
+        dotenv().ok();
+
+        match logger::init_logger() {
+            Ok(_) => info!("Logger has been initialized"),
+            Err(_) => error!("Error logger initialization")
+        };
+
+        //env::set_var("RUST_LOG", "actix_web=debug");
```
Comment (damszew) [inline src/server/builder.rs:125]: leftover

File: src/server/builder.rs [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+        let numthreads = threads::num_threads();
+        info!("Configuring for {} threads", numthreads);
+
+        let app_data = web::Data::new(prepare_app_data());
+        let stats = web::Data::new(BaseStats::default());
+
+        #[allow(unused)]
+        let prod_env = self.run_env == "prod";
+
+        let app_port = self.app_port.to_string();
```
Comment (damszew) [inline src/server/builder.rs:136]: if we are allocating string here - why not just keep `app_port: String` in builder? For ease of use `set_app_port` could accept `ToString` to allow different types etc

Same for other &str fields

**Review by damszew — APPROVED** [review]

File: src/server/builder.rs (hunk 1 above)
Comment (sfisol) [reply src/server/builder.rs:103]: Can't do it, cors_factory is of type `C: Fn() -> Cors + Send + Clone + 'static` while default_cors is of type `fn() -> actix_cors::Cors`

File: src/server/builder.rs (hunk 3 above)
Comment (sfisol) [reply src/server/builder.rs]: This will increase string allocations as default() will at once alloc 3 strings. I just removed this `to_string()` :-)

**Pull request merged**